pyvista/plotting/camera.py

```diff
@@ -691,22 +699,37 @@ def tight(self, padding=0.0, adjust_render_window=True):
         self._renderer.ComputeAspect()
         aspect = self._renderer.GetAspect()
         angle = np.pi * self.view_angle / 180.0
-        dx, dy = (x1 - x0), (y1 - y0)
-        dist = max(dx / aspect[0], dy) / np.sin(angle / 2) / 2
 
-        self.SetViewUp(0, 1, 0)
-        self.SetPosition(x0 + dx / 2, y0 + dy / 2, dist * (1 + padding))
-        self.SetFocalPoint(x0 + dx / 2, y0 + dy / 2, 0)
+        distance_object = np.array([(x1 - x0), (y1 - y0), (z1 - z0)])
+        position = np.array([x0, y0, z0]) + distance_object / 2
+
+        direction, viewup = view_vectors(view, negative)
+        horizontal = np.cross(direction, viewup)
+
+        vert_dist = np.abs(np.dot(distance_object, viewup))
+        horiz_dist = np.abs(np.dot(distance_object, horizontal))
+
+        dist = max(horiz_dist / aspect[0], vert_dist) / np.sin(angle / 2) / 2
+
+        focal_point = position.copy()
+        # set focal point to 0 in viewing direction
+        focal_point -= np.abs(direction * focal_point)
+        # offset position from foral point by dist in viewing direction
+        position += -1 * np.abs(direction) * position + dist * direction
```

Comment:
Again the logic is a bit unclear to me. `abs(direction)` surprises me, because this will in general mess with your direction. I guess what I'm missing here is that this method is special-cased for vectors where exactly one component of any vector is nonzero, and that nonzero value is +-1.

Judging by the preceding comment I'd expect something like
```py
position = focal_point - dist * direction
```

Reply:
> I guess what I'm missing here is that this method is special-cased for vectors where exactly one component of any vector is nonzero, and that nonzero value is +-1.

This is exactly the case here. See my separate comment below for the logic.  I'm open to a more clear implementation, but your comments show at the very least we need to spell out the logic using comments.